storm-server/src/main/java/org/apache/storm/scheduler/resource/ResourceAwareScheduler.java

```diff
@@ -166,7 +166,7 @@ private void scheduleTopology(TopologyDetails td, Cluster cluster, final User to
                     Future<SchedulingResult> schedulingFuture = backgroundScheduling.submit(
                         () -> finalRasStrategy.schedule(toSchedule, td));
                     try {
-                        result = schedulingFuture.get(schedulingTimeoutSeconds, TimeUnit.SECONDS);
+                        result = schedulingFuture.get(schedulingTimeoutSeconds + 1, TimeUnit.SECONDS);
```

Comment:
Why do we need ` + 1`

Reply:
The timeout is check in strategy to determine when terminate. However, if the future task is killed at or around the same time - this results in a TimeOut exception on the task and the result is not propagated back to the caller. +1 gives an additional second before the FutureTask is rudely terminated and allows the result to be returned and examined for the actual message in the result.

Reply:
I think we should discuss about whether we want to do this or not. Essentially this applies to every timeout. If our "timeout=x second" in Storm means things will fail/time out at `x+1` seconds, then everywhere with timeout configs, we need `+1` to make the semantic consistent. I think this is not very necessary

Reply:
Should not apply to every timeout. If the scheduled task is a cooperating task that is also using the same timeout to determine when to stop, then we have this situation where scheduler is interrupting the FutureTask before the task is allowed to gracefully exit and return a result.

If the scheduled task is a non-cooperating task (i.e. .not using the timeout), then it is fine to use the specified number.

Reply:
Probably I misunderstood. Could you point out to me where we are using `SCHEDULING_TIMEOUT_SECONDS_PER_TOPOLOGY` too?  It looks to me here is the only place the schedulingTimeoutSeconds is being used. I don't see cooperations.

Reply:
I see your point. 

When I think about cooperating process, is that the strategy is a time-bound task and part of the same code base running in the same JVM - so there never should be a a need to kill a FutureTask except as a precaution against bug introduced inadvertently.

Current ConstraintSolver uses a different (but redundant) config variable for time limit - which is accidentally set to the same default value.

In light of this - it may be better to explicitly pass "max" time limit to the constraint solver. And then determine how much the margin needs to be, and then add the margin to the FutureTask timeout. Note that this extra margin (and the timeout exception should only happen in exceptional  case when there is bug in ConstraintSolver. Normally it will/should exit by the timeout duration.
And the result should be available.

Reply:
It makes sense to me. Thanks.

Reply:
Can you please add some brief comments about the purpose of `+1` so future me will not be surprised when I come back to this. Thanks

Reply:
Pushed the change into ConstrainstSolverStrategy where there is millisecond granularity and avoid hitting the ceiling. Removed +1 from ResourceAwareStrategy.